developer-hub/src/polkadot_sdk/frame_runtime.rs

```diff
@@ -0,0 +1,140 @@
+//! # FRAME
+//!
+//! ```no_compile
+//!   ______   ______    ________   ___ __ __   ______
+//!  /_____/\ /_____/\  /_______/\ /__//_//_/\ /_____/\
+//!  \::::_\/_\:::_ \ \ \::: _  \ \\::\| \| \ \\::::_\/_
+//!   \:\/___/\\:(_) ) )_\::(_)  \ \\:.      \ \\:\/___/\
+//!    \:::._\/ \: __ `\ \\:: __  \ \\:.\-/\  \ \\::___\/_
+//!     \:\ \    \ \ `\ \ \\:.\ \  \ \\. \  \  \ \\:\____/\
+//!      \_\/     \_\/ \_\/ \__\/\__\/ \__\/ \__\/ \_____\/
+//! ```
+//!
+//! > **F**ramework for **R**untime **A**ggregation of **M**odularized **E**ntities: Substrate's
+//! > State Transition Function (Runtime) Framework.
+//!
+//! ## Introduction
+//!
+//! As described in [`crate::reference_docs::wasm_meta_protocol`], at a high-level substrate-based blockchains are composed of two parts:
+//!
+//! 1. A *runtime* which represents the state transition function (i.e. "Business Logic") of a
+//! blockchain, and is encoded as a Wasm blob.
+//! 2. A client whose primary purpose is to execute the given runtime.
+#![doc = simple_mermaid::mermaid!("../../../docs/mermaid/substrate_simple.mmd")]
+//!
+//! *FRAME is the Substrate's framework of choice to build a runtime.*
+//!
+//! FRAME is composed of two major components, **pallets** and a **runtime**.
+//!
+//! ## Pallets
+//!
+//! A pallet is a unit of encapsulated logic. It has a clearly defined responsibility and can be
+//! linked to other pallets. Each pallet should try to only care about its own responsibilities and
+//! make as few assumptions about the general runtime as possible. A pallet is analogous to a
+//! _module_ in the runtime.
+//!
+//! A pallet is defined as a `mod pallet` wrapped by the [`frame::pallet`] macro. Within this macro,
+//! pallet components/parts can be defined. Most notable of these parts are:
+//!
+//! - [Config](frame::pallet_macros::config), allowing a pallet to make itself configurable and
+//!   generic over types, values and such.
+//! - [Storage](frame::pallet_macros::storage), allowing a pallet to define onchain storage.
+//! - [Dispatchable function aka. Extrinsics](frame::pallet_macros::call), allowing a pallet to
+//!   define extrinsics that are callable by end users, from the outer world.
```

Comment:
I am not in favour of mixing up the words 'dispatchable' and 'extrinsic', I see this repeatedly. Whereas I like the distinction made here https://substrate.stackexchange.com/questions/1754/what-is-the-difference-between-extrinsic-and-dispatchable

Reply:
This is a great suggestion, thanks!